includes/class-omise-money.php

```diff
@@ -0,0 +1,87 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * @since 3.5
+ */
+class Omise_Money {
+	/**
+	 * @var float
+	 */
+	protected $amount;
+
+	/**
+	 * @var string
+	 */
+	protected $currency;
+
+	/**
+	 * @var array
+	 */
+	private $subunit_multiplier = array(
+		'AUD' => 100,
+		'CAD' => 100,
+		'CHF' => 100,
+		'CNY' => 100,
+		'DKK' => 100,
+		'EUR' => 100,
+		'GBP' => 100,
+		'HKD' => 100,
+		'JPY' => 1,
+		'MYR' => 100,
+		'SGD' => 100,
+		'THB' => 100,
+		'USD' => 100
+	);
+
+	/**
+	 * @param int|float|string $amount
+	 * @param string           $currency
+	 */
+	public function __construct( $amount, $currency ) {
+		if ( ! isset( $this->subunit_multiplier[ strtoupper( $currency ) ] ) ) {
+			throw new Exception( __( 'We do not support the currency you are using.', 'omise' ) );
+		}
+
+		$this->amount   = $this->purify_amount( $amount );
+		$this->currency = strtoupper( $currency );
+	}
+
+	/**
+	 * @param  int|float|string $amount
+	 *
+	 * @return float
+	 */
+	public function purify_amount( $amount ) {
+		if ( ! is_string( $amount ) && ! is_float( $amount ) && ! is_numeric( $amount ) ) {
+			throw new Exception( __( 'An amount has to be integer, float, or string.', 'omise' ) );
```

Comment:
Seems a little OTT supporting strings and formatted strings

Reply:
I was thinking about a situation where we are not sure if the platform we are integrating with is returning an amount in a string, float, or int type – which is pretty common in PHP...

What do you say? Better to remove the support of string? or you just wanna leave a note only?
(to me, I prefer to keep it, but you may wanna explain more the downside of this that I didn't see)

Reply:
I just seems like overkill. Especially when this code is being added to a known environment (our woo_commerce plugin) - where we know what to expect. Unless you are advocating for this to be a separate class that we also add in to other projects - in which case we would be better off adding it to `omise-php` as discussed. I really think though, that in the case of units/subunits, all we need is 2 simple static methods on some kind of Omise utility class

Reply:
Right, I was designing this method with a thought of "this will be used all over projects".
Well, agreed, `An amount has to be string` seems weird. Removing